deployer/models/release_safe.go

```diff
@@ -60,122 +60,194 @@ func (release *Release) ValidateSafeRelease(s3c aws.S3API, resources *ReleaseRes
 	// Set Defaults for comparison
 	previousRelease.Release.SetDefaults(release.AwsRegion, release.AwsAccountID, "coinbase-odin-")
 	previousRelease.SetDefaults()
+
+	// return an error for valid services
 	return release.validateSafeRelease(&previousRelease)
 }
 
+type SafeReleaseError struct {
+	Subnets        error
+	Timeout        error
+	AllServices    error
+	MissingService error
+
+	Services map[string]*SafeReleaseServiceError
+}
+
+type SafeReleaseServiceError struct {
```

Comment:
not sure i get the advantage of breaking down the errors to different fields. imo just having SafeReleaseServiceError be a typedef for error would make more sense.

Maybe having a separate subclass for each error type if we need to get the exact type later on

Reply:
nevermind i think this makes sense now

Reply:
This is also in part for a future feature where I want to let a user ignore a particular field. At the moment this does not contain `Image` because it often changes. But I would like to have the ability to check the Image, but ignore it. So we can have by default safe_release on, with varying degrees of safety.